### ISSUE TITLE
Potential unexpected behavior of commission and PL calculated

### ISSUE THREAD

**Issue by exodus713** [nautechsystems/nautilus_trader#657]
# Bug Report

The commission total amount shown by the platform as well as the PL does not seem to correspond to what was expected.

This might not be a bug at all but just a misunderstanding of the platform behavior from me...

### Expected Behavior
we would expect PL = approximately Commission from the backtest report as we have set the commission to 1% - prohibitive for testing - and the strategy trades a lot. 

we would expect Commission from report = sum(abs(notional traded)) * 1%. = check_commission_amount in the notebook

### Steps to Reproduce the Problem

code and data to reproduce the example are attached.

[issue_git_export.zip](https://github.com/nautechsystems/nautilus_trader/files/8779958/issue_git_export.zip)

### Specifications

- OS platform: Ubuntu 20.04.4
- Python version: 3.8.10
- `nautilus_trader` version: 1.146

**Comment by cjdsellers** [comment]
Thank you for the report and for providing the code with data to reproduce.

Will look into this over the weekend and get back to you on my findings.

**Comment by cjdsellers** [comment]
I've managed to run your test script. You might have had a reason although you could probably set the `taker_fee` to a positive `0.01`. When I do this the value calculated by your test becomes `16219.56290000` vs  `16_759.14639500` for the backtest result (which is much closer, but not exact).

The area to look at the commission calculations would start at `backtest/exchange.pyx` L1568
```
    cdef void _fill_order(
        self,
        Instrument instrument,
        Order order,
        PositionId venue_position_id,
        Position position,  # Can be None
        Quantity last_qty,
        Price last_px,
        LiquiditySide liquidity_side,
    ) except *:
        # Calculate commission
        cdef Money commission = self.exec_client.get_account().calculate_commission(
            instrument=instrument,
            last_qty=order.quantity,
            last_px=last_px,
            liquidity_side=liquidity_side,
        )
```

and the `calculate_commission` methods for each account type:
- `accounting/accounts/cash.pyx` 
- `accounting/accounts/margin.pyx`

It indeed does look like a simple `notional * fee` type calculation:
```
        if liquidity_side == LiquiditySide.MAKER:
            commission = notional * float(instrument.maker_fee)
        elif liquidity_side == LiquiditySide.TAKER:
            commission = notional * float(instrument.taker_fee)
        else:  # pragma: no cover (design-time error)
            raise ValueError(
                f"invalid LiquiditySide, was {LiquiditySideParser.to_str(liquidity_side)}"
            )

        if instrument.is_inverse and not inverse_as_quote:
            return Money(commission, instrument.base_currency)
        else:
            return Money(commission, instrument.quote_currency)
```

However always in terms of quote currency, which may not be what you expect (I believe Binance uses base currency for buy, and quote currency for sell?).

I'm not sure the maths really works to take the cumulative fills post backtest and multiply by the taker fee?

Were you able to discover anything else? If you're able to provide a `.ipynb` I can keep digging.

**Comment by exodus713** [comment]
First of all thank you very much Chistopher for this.

I've looked at the functions that you showed and they look indeed ok.
Unfortunately it is very difficult to check the whole flow of the code when
there is an execution / fill as it is in Cython and I've not found a
convenient way to debug it along with python.

Nevertheless I can say that there is clearly something unexpected in the
computation of the PL & commissions.
I've run tests with 10% commissions, both taker and maker and the loss is
approximately 2 times what would be expected i.e. 2 times the commission
amount.
And yes commission should be commission rate * sum (abs(notional traded))
but it is not the case.

In summary 2 issues :
- PL when large commission = 2 times commissions. which seems to indicate
that commissions are double counted somewhere. I suspect it is the cash
account impact of the trade & commissions
- Commission shown in the report is not equal to expected commissions.

Sorry to insist on that but it is very important to make sure that the tool
shows 100% accurate commission computations as this is a key element of
expected PL in real life.

Do not hesitate to let me know if I can help.

**Comment by cjdsellers** [comment]
Isn't commission charged for each side of the round trip though? Each trade will attract the `notional * fee` commission?

**Comment by exodus713** [comment]
Yes indeed. That is why i took sum(abs(notionnal *fee) in my computation. And still have a PL that is twice as expected.

**Comment by cjdsellers** [comment]
I've traced the flows again and can't seem to find any issue or double counting.

Could this be a case of simply how PnL and commissions are presented in the report?

PnL includes commissions, then commissions are also shown separately.

**Comment by exodus713** [comment]
Sorry for the late reply as I was travelling.
I am pretty sure that this is not a presentation issue unfortunately.
But I am happy to go through the example with you on the phone and explain
where the mismatch is, in my opinion, if this could be of help to you.

**Comment by cjdsellers** [comment]
Could you please run your test script again when you have a moment. 

There was a bug uncovered for position snapshots which has now been fixed https://github.com/nautechsystems/nautilus_trader/issues/719.

**Comment by cjdsellers** [comment]
Hi @exodus713

It turns out you were entirely correct. Commissions have been getting double counted for both single and multi currency accounts.
The commission was being subtracted once in the PnL calculation, and then again in the account balance update.

This has now been rectified with ddee268b7dda9abf33c286b380ff6a1fbb2f2e0f on `develop`.

**Comment by exodus713** [comment]
Yes exactly. That is what i am expecting.
And why it might not match unless I am missing something.
